### PR TITLE
[RHCLOUD-24402] Adding extra schema properties

### PR DESCRIPTION
**Pull request by tahmidefaz**
This is how these properties look like. This is just an example.

image

### CODE REVIEW

File: src/components/APIDoc/SchemaDataView.tsx [hunk 1]
```diff
@@ -56,15 +61,80 @@ const PropertyView:React.FunctionComponent<PropertyComponentProps> = ({propName,
...
+      {propSchema.multipleOf && <Label isCompact>multipleOf: {propSchema.multipleOf}</Label>}
+      {maxMin && <Label isCompact>{maxMin}</Label>}
+      {maxMinChar && <Label isCompact>{maxMinChar}</Label>}
+      {maxMinItems && <Label isCompact>{maxMinItems}</Label>}
+      {maxMinProps && <Label isCompact>{maxMinProps}</Label>}
+      {propSchema.uniqueItems && <Label isCompact>unique</Label>}
+      {propSchema.nullable && <Label isCompact>nullable</Label>}
+      {propSchema.readOnly && <Label isCompact>read only</Label>}
+      {propSchema.writeOnly && <Label isCompact>write only</Label>}
+      {propSchema.deprecated && <Label isCompact>depricated</Label>}
```
Comment (josejulio) [inline src/components/APIDoc/SchemaDataView.tsx:133]: typo
```suggestion
      {propSchema.deprecated && <Label isCompact>deprecated</Label>}
```

File: src/components/APIDoc/SchemaDataView.tsx [hunk 2]
```diff
@@ -56,15 +61,80 @@ const PropertyView:React.FunctionComponent<PropertyComponentProps> = ({propName,
...
+  if (propSchema.maxProperties && propSchema.minProperties) {
+    maxMinProps = `${propSchema.minProperties} and ${propSchema.maxProperties} properties`
+  } else if (propSchema.maxProperties) {
+    maxMinProps = `max ${propSchema.maxProperties} properties`
+  } else if (propSchema.minItems) {
+    maxMinProps = `min ${propSchema.minProperties} properties`
+  }
+
+  return(
+    <TextContent>
```
Comment (josejulio) [inline src/components/APIDoc/SchemaDataView.tsx:119]: Could you use a `LabelGroup` instead? No special props are needed

https://patternfly-react.surge.sh/components/label-group

File: src/components/APIDoc/SchemaDataView.tsx [hunk 3]
```diff
@@ -56,15 +61,80 @@ const PropertyView:React.FunctionComponent<PropertyComponentProps> = ({propName,
...
       </FlexItem>
     </Flex>
   )
 }
 
+
+interface ExtraPropertyViewProps {
+  propSchema: OpenAPIV3.SchemaObject;
+}
+const ExtraPropertyView:React.FunctionComponent<ExtraPropertyViewProps> = ({propSchema}) => {
```
Comment (josejulio) [inline src/components/APIDoc/SchemaDataView.tsx:80]: We could move it to a separate file, but we could do that later.

File: src/components/APIDoc/SchemaDataView.tsx [hunk 4]
```diff
@@ -56,15 +61,80 @@ const PropertyView:React.FunctionComponent<PropertyComponentProps> = ({propName,
...
+    maxMin += (propSchema.exclusiveMaximum ? '<' : '≤') + ` ${propSchema.maximum}`
+  } else if (propSchema.maximum) {
+    maxMin = (propSchema.exclusiveMaximum ? '<' : '≤') + ` ${propSchema.maximum}`
+  } else if (propSchema.minimum) {
+    maxMin = (propSchema.exclusiveMinimum ? '>' : '≥') + ` ${propSchema.minimum}`
+  }
+
+  let maxMinChar: string | undefined;
+  if (propSchema.maxLength && propSchema.minLength) {
+    maxMinChar = `${propSchema.minLength} to ${propSchema.maxLength} chars`
```
Comment (josejulio) [inline src/components/APIDoc/SchemaDataView.tsx:93]: The screenshot renders as `7 to 3 chars` and I'm like ... min is 7 and max is 3??

File: src/components/APIDoc/SchemaDataView.tsx [hunk 5]
```diff
@@ -56,15 +61,80 @@ const PropertyView:React.FunctionComponent<PropertyComponentProps> = ({propName,
...
+    maxMinProps = `max ${propSchema.maxProperties} properties`
+  } else if (propSchema.minItems) {
+    maxMinProps = `min ${propSchema.minProperties} properties`
+  }
+
+  return(
+    <TextContent>
+      {propSchema.format && <Label isCompact>{propSchema.format}</Label>}
+      {propSchema.default && <Label isCompact>default: {propSchema.default}</Label>}
+      {propSchema.enum && <Label isCompact>{propSchema.enum.join(" | ")}</Label>}
```
Comment (josejulio) [inline src/components/APIDoc/SchemaDataView.tsx:122]: We should probably have "one of " or "enums:" - only the values is confusing

File: src/components/APIDoc/SchemaDataView.tsx (hunk 3 above)
Comment (tahmidefaz) [reply src/components/APIDoc/SchemaDataView.tsx]: I ended up moving the whole PropertyView, including ExtraPropertyView to a separate file.

File: src/components/APIDoc/SchemaDataView.tsx (hunk 4 above)
Comment (tahmidefaz) [reply src/components/APIDoc/SchemaDataView.tsx]: Whoops, human error :sweat_smile:

File: src/components/APIDoc/SchemaDataView.tsx (hunk 5 above)
Comment (tahmidefaz) [reply src/components/APIDoc/SchemaDataView.tsx]: Done! Choose enums such as not to confuse people with the `oneOf` property

**Review by josejulio — APPROVED** [review]
lgtm

**Pull request merged**